### PR TITLE
The change in the measurement of CVAT-UI code coverage.

### PR DESCRIPTION
**Pull request by dvkruchinin**
Update Cypress to 5.3 (Firefox support is out of beta)
The use of the plugin "@cypress/code-coverage" for running tests after code instrumentations. 

### Motivation and context
Speed up passing tests after code instrumentation using the Firefox browser. This change made it possible to reduce the launch time of each test by more than 3 times. In addition, don't need to run tests via Google Chrome to measure coverage.

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -33,29 +33,23 @@ before_script:
   - chmod a+rwx ${HOST_COVERAGE_DATA_DIR}
 
 script:
-  - docker-compose -f docker-compose.yml -f docker-compose.ci.yml run cvat_ci /bin/bash -c 'coverage run -a manage.py test cvat/apps utils/cli && mv .coverage ${CONTAINER_COVERAGE_DATA_DIR}'
```
Comment (bsekachev) [inline .travis.yml:36]: Can we leave it as is?

File: tests/package.json [hunk 2]
```diff
@@ -1,12 +1,12 @@
 {
   "scripts": {
-    "cypress:run:chrome": "cypress run --headless --browser chrome",
+    "cypress:run:chrome": "cypress run --env coverage=false --headless --browser chrome",
```
Comment (bsekachev) [inline tests/package.json:3]: Is this enabled by default?
Can we disable coverage for Firefox by default? (by passing the same argument) And override it in .travis.yml?

Comment (bsekachev) [reply tests/package.json:3]: Do you use this arg in the code explicitly?

File: .travis.yml [hunk 3]
```diff
@@ -33,29 +33,23 @@ before_script:
...
       npm run report:coverage:lcov;
     else
       cd ./tests && npm install &&
       npm run cypress:run:chrome && cd ..;
     fi;
-
-
-after_success:
-  # https://coveralls-python.readthedocs.io/en/latest/usage/multilang.html
-  - docker-compose -f docker-compose.yml -f docker-compose.ci.yml run cvat_ci /bin/bash -c 'ln -s ${CONTAINER_COVERAGE_DATA_DIR}/.git . && ln -s ${CONTAINER_COVERAGE_DATA_DIR}/.coverage . && ln -s ${CONTAINER_COVERAGE_DATA_DIR}/coverage.json . && coveralls --merge=coverage.json'
```
Comment (bsekachev) [inline .travis.yml:61]: I've just realized that the coverage is computed also for a server part. Could you please investigate if we could have two different badges for client and for server? If it is impossible, anyway we need left coverage for server part as is and merge results for the same badge (maybe also we should to run server coverage only on cron builds)

File: .travis.yml (hunk 1 above)
Comment (dvkruchinin) [reply .travis.yml:36]: In this case, the coverage measurement readings will alternate depending on the type of CI being triggered. This command also sends data to the coveralls server.

File: tests/package.json (hunk 2 above)
Comment (dvkruchinin) [reply tests/package.json:3]: Yes, coverage measurement is enabled by default in the plugin.
Alternatively, you can make 2 commands for each launch. With and without coverage measurement.
I use it as an argument for running through npm.

File: .travis.yml (hunk 3 above)
Comment (dvkruchinin) [reply .travis.yml:61]: Well. I will investigate this topic. It will probably be easier to combine the results.

**Comment by dvkruchinin** [comment]
Redesigned the logic for running tests and measuring coverage. In cron, run tests via Firefox without measuring coverage. For the rest, run tests via Chrome with coverage measurement. Combining coverage measurement results for core and 2e2 tests.

**Comment by bsekachev** [comment]
@dvkruchinin 
Please, let me know when the PR is ready. Just remove WIP tag

**Comment by dvkruchinin** [comment]
> @dvkruchinin
> Please, let me know when the PR is ready. Just remove WIP tag

I got a crash error for the first test after rebuilding containers after code instrumentation. When logging in, a modal window appeared with the text "Could not get user agreements from the server". To avoid this, I added a 10-second wait to the first test.

**Review by bsekachev — APPROVED** [review]

**Pull request merged**